### PR TITLE
Maya/194962 194961 gisaid genbank api

### PR DESCRIPTION
**Pull request by mmmmmmaya**
### Summary
- **What:** Genericize file downloads from the samples table
- **Why:** Support for gisaid and genbank templates
- **Ticket:**
  - [[sc-194961]](https://app.shortcut.com/genepi/story/194961)
  - [[sc-194962]](https://app.shortcut.com/genepi/story/194962)
- **Env:** https://maya-api-frontend.dev.czgenepi.org/

### Testing
1. Try downloading fasta and metadata files to make sure they still work
1. Gisaid and genbank downloads don't actually return anything yet

### Demos
No visual changes.

### Notes
This will need a small amount of work once the BE endpoints are in place to handle the case where multiple files need to be downloaded for GISAID/Genbank templates. I don't know how to handle that without having endpoint access. I think this is ok to merge, since the fasta and metadata downloads both still work as expected, and the other two options are behind a feature flag.

### Checklist
- [X] I merged latest `trunk`
- [x] I manually verified the change
- [x] I added labels to my PR
- [x] I either asked design for a review or hid my changes behind a feature flag
- [ ] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/frontend/src/common/analytics/eventTypes.ts [hunk 1]
```diff
@@ -169,6 +169,10 @@ export type AnalyticsSamplesDownloadFile = {
   sample_public_ids: JsonString;
   // User downloaded info on consensus genome (FASTA) for these samples
   includes_consensus_genome: boolean;
+  // User downloaded genbank template for these samples
```
Comment (mmmmmmaya) [inline src/frontend/src/common/analytics/eventTypes.ts:172]: @vincent-czi would like to make sure this is an ok change/what I should do

Comment (vincent-czi) [reply src/frontend/src/common/analytics/eventTypes.ts:172]: Perfect: this change and the other one to pass te data looks great!

File: src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx [hunk 2]
```diff
@@ -244,8 +255,11 @@ const DownloadModal = ({
                     <DownloadTypeInfo>
                       Download concatenated consensus genomes and metadata files
                       formatted to prepare samples for submission to GISAID.{" "}
-                      {/* TODO: (194961) - update href */}
-                      <Link href="" target="_blank" rel="noreferrer">
+                      <Link
+                        href="https://help.czgenepi.org/hc/en-us/articles/8179880474260"
```
Comment (ehoops-cz) [inline src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx:259]: 🙇‍♀️

File: src/frontend/src/common/queries/samples.ts [hunk 3]
```diff
@@ -24,42 +24,43 @@ import { MutationCallbacks } from "./types";
 /**
  * Download fasta file for samples
  */
-interface SampleFastaDownloadPayload {
+
+type FileDownloadEndpointType =
```
Comment (ehoops-cz) [inline src/frontend/src/common/queries/samples.ts:28]: nit: do these names need 'type' in them?  Seems redundant.

**Review by ehoops-cz — APPROVED** [review]
Looks good!

File: src/frontend/src/common/queries/samples.ts (hunk 3 above)
Comment (mmmmmmaya) [reply src/frontend/src/common/queries/samples.ts:28]: I prefer for types and interfaces to end with Type so it's clear which vars are types and which aren't. It's a pattern we have throughout the app

**Pull request merged**